src/Uno.UI.RemoteControl/HotReload/ClientHotReloadProcessor.Agent.cs

```diff
@@ -0,0 +1,192 @@
+		/// <summary>
+		/// Waits for the server's hot reload workspace to be loaded
+		/// </summary>
+		/// <param name="ct"></param>
+		/// <returns></returns>
+		public Task<bool> WaitForWorkspaceLoaded(CancellationToken ct)
+			=> _hotReloadWorkloadSpaceLoaded.Task;
+
+		[MemberNotNull(nameof(_agent))]
+		partial void InitializeMetadataUpdater()
+		{
+			_instance = this;
+
+			_linkerEnabled = string.Equals(Environment.GetEnvironmentVariable("UNO_BOOTSTRAP_LINKER_ENABLED"), "true", StringComparison.OrdinalIgnoreCase);
+
+			if (_linkerEnabled)
+			{
+				var message = "The application was compiled with the IL linker enabled, hot reload is disabled. " +
+							"See WasmShellILLinkerEnabled for more details.";
+
+				Console.WriteLine($"[ERROR] {message}");
+			}
+
+			_agent = new HotReloadAgent(s =>
+			{
+				if (this.Log().IsEnabled(LogLevel.Trace))
+				{
+					this.Log().Trace(s);
+				}
+			});
+		}
+
+
+		private bool MetadataUpdatesEnabled
+		{
+			get
+			{
+				var unoRuntimeIdentifier = GetMSBuildProperty("UnoRuntimeIdentifier");
+				var targetFramework = GetMSBuildProperty("TargetFramework");
+				var buildingInsideVisualStudio = GetMSBuildProperty("BuildingInsideVisualStudio");
+
+				return
+					buildingInsideVisualStudio.Equals("true", StringComparison.OrdinalIgnoreCase)
+					&& (
+						// As of VS 17.8, when the debugger is not attached, mobile targets can use
+						// DevServer's hotreload workspace, as visual studio does not enable it on its own.
+						(!Debugger.IsAttached
+							&& (targetFramework.Contains("-android") || targetFramework.Contains("-ios"))));
+			}
+		}
+
+		private string[] GetMetadataUpdateCapabilities()
+		{
+			if (Type.GetType(HotReloadAgent.MetadataUpdaterType) is { } type)
+			{
+				if (type.GetMethod("GetCapabilities", BindingFlags.Static | BindingFlags.NonPublic) is { } getCapabilities)
+				{
+					if (getCapabilities.Invoke(null, Array.Empty<string>()) is string caps)
+					{
+						if (this.Log().IsEnabled(LogLevel.Trace))
+						{
+							this.Log().Trace($"Metadata Updates runtime capabilities: {caps}");
+						}
+
+						return caps.Split(' ');
+					}
+					else
+					{
+						if (this.Log().IsEnabled(LogLevel.Warning))
+						{
+							this.Log().Trace($"Runtime does not support Hot Reload (Invalid returned type for {HotReloadAgent.MetadataUpdaterType}.GetCapabilities())");
+						}
+					}
+				}
+				else
+				{
+					if (this.Log().IsEnabled(LogLevel.Warning))
+					{
+						this.Log().Trace($"Runtime does not support Hot Reload (Unable to find method {HotReloadAgent.MetadataUpdaterType}.GetCapabilities())");
+					}
+				}
+			}
+			else
+			{
+				if (this.Log().IsEnabled(LogLevel.Warning))
+				{
+					this.Log().Trace($"Runtime does not support Hot Reload (Unable to find type {HotReloadAgent.MetadataUpdaterType})");
+				}
+			}
+			return Array.Empty<string>();
+		}
+
+#if __WASM__ || __SKIA__
```

Comment:
This has been changed in a merged PR, it's probably best if you re-do this change to align.

Reply:
Actually, the update was lost somehow... it's very strange, I'll change it again in a separate PR.

Reply:
rebased and picked up latest changes